crates/types/src/blockchain/header.rs

```diff
@@ -441,6 +441,7 @@ impl ApplicationHeader<GeneratedApplicationFields> {
         hasher.input(self.message_receipt_count.to_be_bytes());
         hasher.input(self.transactions_root.as_ref());
         hasher.input(self.message_outbox_root.as_ref());
+        hasher.input(self.event_inbox_root.as_ref());
```

Comment:
Is there any logic to the order these things get added to the hash?

Reply:
The order of fields in the header